### PR TITLE
PROD-4121 : Allow members to manage their notification subscriptions

### PR DESCRIPTION
**Pull request by milanhirapra**
### Jira Issue:

https://buddyboss.atlassian.net/browse/PROD-4121
https://buddyboss.atlassian.net/browse/PROD-4239

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

File: src/bp-forums/core/filters.php [hunk 1]
```diff
@@ -318,3 +318,34 @@ function _bbp_has_replies_query( $args = array() ) {
...
+ *
+ * @param string  $new_status New post status.
+ * @param string  $old_status Old post status.
+ * @param WP_Post $post       Post object.
+ */
+function bb_forums_update_subscription_status( $new_status, $old_status, $post ) {
+	if ( $new_status !== $old_status && ! empty( $post->post_type ) && in_array( $post->post_type, array( bbp_get_forum_post_type(), bbp_get_topic_post_type() ), true ) ) {
+
+		// Get current site ID.
+		$blog_id = bp_get_root_blog_id();
```
Comment (KartikSuthar) [inline src/bp-forums/core/filters.php:335]: remove this its not needed

File: src/bp-forums/core/filters.php [hunk 2]
```diff
@@ -318,3 +318,34 @@ function _bbp_has_replies_query( $args = array() ) {
...
+		if ( ! empty( $new_status ) && in_array( $new_status, array( bbp_get_spam_status_id(), bbp_get_trash_status_id(), bbp_get_pending_status_id() ), true ) ) {
+			$subscription_status = 0;
+		}
+
+		$subscription_type = 'topic';
+		if ( bbp_get_forum_post_type() === $post->post_type ) {
+			$subscription_type = 'forum';
+		}
+
+		bb_subscriptions_update_subscriptions_status( $subscription_type, $post->ID, $subscription_status, $blog_id );
```
Comment (KartikSuthar) [inline src/bp-forums/core/filters.php:347]:
```suggestion
		bb_subscriptions_update_subscriptions_status( $subscription_type, $post->ID, $subscription_status );
```
$blog_id is optional param not needed to pass while update

File: src/bp-forums/templates/default/bbpress/form-reply.php [hunk 3]
```diff
@@ -108,13 +108,21 @@
 
 								<?php if ( bbp_is_reply_edit() && ( bbp_get_reply_author_id() !== bbp_get_current_user_id() ) ) : ?>
 
-									<label for="bbp_topic_subscription"><?php _e( 'Notify the author of follow-up replies via email', 'buddyboss' ); ?></label>
+									<label for="bbp_topic_subscription"><?php esc_html_e( 'Notify the author of follow-up replies via email', 'buddyboss' ); ?></label>
```
Comment (KartikSuthar) [inline src/bp-forums/templates/default/bbpress/form-reply.php:111]: update the same condition here for the legacy email and the new one.

Do same for the topic as well

File: src/bp-notifications/bp-notifications-settings.php [hunk 4]
```diff
@@ -451,11 +458,35 @@ function bb_admin_setting_callback_notification_warning() {
...
+	if ( ! empty( $field['notification_tooltip_text'] ) ) {
+		$tooltip_pos  = 'up';
+		$tooltip_text = $field['notification_tooltip_text'];
+	}
+	$disabled = ! empty( $field['notification_read_only'] ) && true === $field['notification_read_only'];
 	?>
 
 	<input name="bb_enabled_notification[<?php echo esc_attr( $field['key'] ); ?>][main]" type="hidden" value="no" />
-	<input class="bb-notification-checkbox" id="bb_enabled_notification_<?php echo esc_attr( $field['key'] ); ?>" name="bb_enabled_notification[<?php echo esc_attr( $field['key'] ); ?>][main]" type="checkbox" value="yes" <?php checked( $checked, 1 ); ?> />
+	<span class="forum-subscription-input"
```
Comment (KartikSuthar) [inline src/bp-notifications/bp-notifications-settings.php:472]: Change the class name to target globally for all, not for the subscription specific.

File: src/bp-templates/bp-nouveau/buddypress/members/single/settings/subscriptions.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * The template for members settings ( Subscription )
+ *
+ * This template can be overridden by copying it to yourtheme/buddypress/members/single/settings/subscriptions.php.
+ *
+ * @since BuddyBoss [BBVERSION]
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/members/single/settings/subscriptions.php:7]: Add version here

File: src/bp-templates/bp-nouveau/buddypress/members/single/parts/notification-subnav.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * BuddyBoss Account Notification Navigation
+ *
+ * This template can be overridden by copying it to yourtheme/buddypress/members/single/parts/notification-subnav.php.
+ *
+ * @since BuddyBoss [BBVERSION]
+ */
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/buddypress/members/single/parts/notification-subnav.php:8]: Add version here

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Review by KartikSuthar — DISMISSED** [review]

**Review by jitendrabanjara1991 — APPROVED** [review]

**Review by KartikSuthar — DISMISSED** [review]

**Review by KartikSuthar — DISMISSED** [review]

**Review by KartikSuthar — DISMISSED** [review]

**Review by KartikSuthar — DISMISSED** [review]

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**